libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/DirectionsSession.kt

```diff
@@ -3,23 +3,17 @@ package com.mapbox.navigation.core.directions.session
 import com.mapbox.api.directions.v5.models.RouteOptions
 import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.base.route.NavigationRouterCallback
-import com.mapbox.navigation.base.route.Router
 import com.mapbox.navigation.core.SetRoutes
 
 internal interface DirectionsSession : RouteRefresh {
 
-    /**
-     * Routes that were fetched from [Router] or set manually.
-     * On [routes] change notify registered [RoutesObserver]
-     *
-     * @see [registerRoutesObserver]
-     */
-    val routes: List<NavigationRoute>
+    val routesUpdatedResult: RoutesUpdatedResult?
 
     val initialLegIndex: Int
 
     fun setRoutes(
         routes: List<NavigationRoute>,
+        ignoredRoutes: List<IgnoredRoute>,
```

Comment:
maybe it should accept `NativeSetRouteResult` as direction session is always updated after updating routes in native?

Reply:
`NativeSetRouteValue` is processed in a place of usage. The question is where this processing should be done. It's done in `MapboxNavigation` now because it is used later (ignoredAlternatives are passed to `RoutesSetCallback`). So we would still need to to the processing there.

Reply:
I notices that you had to update existing unit test because the method's signature was change. Do you think it makes sense to pass an object so that future updates will be easier for us?

I think that it would be great to move the processing away from the `MapboxNavigaiton` so that it acts more as a facade. Maybe `NativeSetRouteValue` could have `ignoredAlternatives` field?

Reply:
Passing an object instead of many parameters could also help to create such an API which avoid situations which doesn't make sense, for example 
```
setRoute(
    routes = emptyList(),
    ignoredRoutes = listOf(route1, route2),
    reason = someReason
)
```
With sealed we could provide API which is harder to use incorrectly. 
```
setRoutes(SetRoutesParam.CleanupRoutes)
```

Reply:
> I notices that you had to update existing unit test because the method's signature was change. Do you think it makes sense to pass an object so that future updates will be easier for us?

How will that help? We'll have to update all those places as well, just instead of passing a new parameter to the method we'll have to pass it to the constructor.

> Passing an object instead of many parameters could also help to create such an API which avoid situations which doesn't make sense

In the example you provided for a reason "new" or "refresh" or anything else except "cleanup" creating an object won't help unless we do an explicit assert.

> I think that it would be great to move the processing away from the MapboxNavigaiton so that it acts more as a facade.

That I agree with. It's not been just a facade for quite a long time now but I'll try not to introduce any more heavy logic there. I'll move that out.

> Maybe NativeSetRouteValue could have ignoredAlternatives field?

The name implies that it's about what NN returned, I don't want to do any additional processing just there.

Reply:
> How will that help? We'll have to update all those places as well, just instead of passing a new parameter to the method we'll have to pass it to the constructor.

You can create objects in tests using factories so that if a new filed added, you just add default value to the factory and all tests are starting to use 

> The name implies that it's about what NN returned

I don't insist on reusing the object, it can be a new one for direction session, I just this that result from trip session is the only reason routes are updated, so maybe it make sense to use the same object.